src/utils/price_comparison_utils.ts

```diff
@@ -0,0 +1,110 @@
+import { BridgeReportSource, ERC20BridgeSource, MultiHopReportSource, QuoteReportSource } from '@0x/asset-swapper';
+import { BigNumber } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import * as _ from 'lodash';
+
+import { GAS_SCHEDULE_V1 } from '../config';
+import { ZERO } from '../constants';
+import { logger } from '../logger';
+import { ChainId, SourceComparison } from '../types';
+
+import { getTokenMetadataIfExists } from './token_metadata_utils';
+
+const renameIfNativeSource = (source: ERC20BridgeSource) => (source === ERC20BridgeSource.Native ? '0x' : source);
+
+const emptyPlaceholderSources = Object.values(ERC20BridgeSource).reduce<SourceComparison[]>((memo, liquiditySource) => {
+    memo.push({
+        name: renameIfNativeSource(liquiditySource),
+        price: null,
+        gas: null,
+    });
+
+    return memo;
+}, []);
+
+interface PartialRequestParams {
+    buyAmount?: BigNumber;
+    sellAmount?: BigNumber;
+}
+
+interface PartialQuote {
+    buyAmount: BigNumber;
+    sellAmount: BigNumber;
+    buyTokenAddress: string;
+    sellTokenAddress: string;
+    quoteReport?: { sourcesConsidered: QuoteReportSource[] };
+}
+
+export const priceComparisonUtils = {
+    getPriceComparisonFromQuote(
+        chainId: ChainId,
+        params: PartialRequestParams,
+        quote: PartialQuote,
+    ): SourceComparison[] | undefined {
+        if (!quote.quoteReport) {
```

Comment:
could we just make `quoteReport` required in `PartialQuote`, and not call this fn if `quoteReport` is `undefined`?